File: module-common/src/main/java/maple/expectation/global/error/exception/SenderMemberNotFoundException.java
package maple.expectation.global.error.exception;

import maple.expectation.global.error.CommonErrorCode;
import maple.expectation.global.error.exception.base.ClientBaseException;
import maple.expectation.global.error.exception.marker.CircuitBreakerIgnoreMarker;

/**
 * 발신자(Sender)의 Member 계정이 DB에 존재하지 않을 때 발생
 *
 * <p>포인트 이체 시 발신자 UUID로 Member를 조회할 수 없을 때 발생합니다.
 */
public class SenderMemberNotFoundException extends ClientBaseException
    implements CircuitBreakerIgnoreMarker {

  public SenderMemberNotFoundException(String maskedUuid) {
    super(CommonErrorCode.SENDER_MEMBER_NOT_FOUND, maskedUuid);
  }
}

File: module-common/src/main/kotlin/maple/expectation/error/exception/SenderMemberNotFoundException.kt
package maple.expectation.error.exception

import maple.expectation.error.CommonErrorCode
import maple.expectation.error.exception.base.ClientBaseException

class SenderMemberNotFoundException(senderId: String) : ClientBaseException(
    CommonErrorCode.SENDER_MEMBER_NOT_FOUND,
    senderId
)
